docs/get-started/clinc150/index.md

```diff
@@ -0,0 +1,843 @@
+provided you have an active account:
+
+```bash
+pip install wandb
+wandb login
+```
+
+Let's start fine-tuning!
+
+```python
+import math
+
+import finetuner
+from finetuner.tuner.callback import (
+    BestModelCheckpoint,
+    EarlyStopping,
+    EvaluationCallback,
+    WandBLogger,
+)
+from finetuner.tuner.pytorch.losses import TripletLoss
+from finetuner.tuner.pytorch.miner import TripletEasyHardMiner
+from transformers.optimization import get_linear_schedule_with_warmup
+
+EPOCHS = 6
+BATCH_SIZE = 256
+LEARNING_RATE = 1e-4
+NUM_WORKERS = 8
+NUM_ITEMS_PER_CLASS = 4
+DEVICE = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+def configure_optimizer(model: torch.nn.Module):
+    optimizer = torch.optim.AdamW(model.parameters(), lr=LEARNING_RATE)
+    scheduler = get_linear_schedule_with_warmup(
+        optimizer,
+        num_warmup_steps=50,
+        num_training_steps=EPOCHS * math.ceil(len(train_data) / BATCH_SIZE),
+    )
+    return optimizer, scheduler
+
+evaluation_callback = EvaluationCallback(
+    val_data, limit=int(num_utts_per_intent_val), num_workers=NUM_WORKERS
+)
+wandb_logger = WandBLogger()
+early_stopping = EarlyStopping(patience=1, monitor='average_precision')
+best_model_ckpt = BestModelCheckpoint(
+    save_dir='checkpoints', monitor='average_precision'
+)
+
+finetuned_model = finetuner.fit(
+    TransformerEmbedder(),
+    train_data=train_data,
+    eval_data=val_data,
+    loss=TripletLoss(
+        distance='cosine',
+        margin=0.4,
+        miner=TripletEasyHardMiner(pos_strategy='easy', neg_strategy='hard'),
+    ),
+    configure_optimizer=configure_optimizer,
+    epochs=EPOCHS,
+    batch_size=BATCH_SIZE,
+    learning_rate=LEARNING_RATE,
+    num_items_per_class=NUM_ITEMS_PER_CLASS,
+    device=DEVICE,
+    collate_fn=collate_fn,
+    callbacks=[evaluation_callback, wandb_logger, early_stopping, best_model_ckpt],
+)
+```
+
+Let's go through the weights and biases run for various training stats. Below is our
+learning rate schedule, our training and validation loss and some evaluation metrics
+calculcated in our val split:
+
+```{figure} wandb01.png
+```
+```{figure} wandb02.png
+```
+
+Now it's time to see how much we improved. Let's re-run the evaluation on our test data,
+this time using the tuned model:
+
+```python
+evaluator = Evaluator(test_data, embed_model=finetuned_model)
+metrics = evaluator.evaluate(
+    limit=int(num_utts_per_intent_test),
+    num_workers=NUM_WORKERS,
+    batch_size=BATCH_SIZE,
+    device=DEVICE,
+    collate_fn=collate_fn,
+)
+print('Evaluating FINE-TUNED model on test data:')
+print(json.dumps(metrics, indent=2))
+```
+```
+Evaluating FINE-TUNED model on test data:
+{
+  "r_precision": 0.9152863975803448,
+  "precision_at_k": 0.8819923371647765,
+  "recall_at_k": 0.8819923371647765,
+  "f1_score_at_k": 0.8819923371647765,
+  "average_precision": 0.9500754908924715,
+  "hit_at_k": 0.9917777777777778,
```

Comment:
i would omit the evaluation print, but turn the results into a table (pretrained vs finetuned), and as Tadej said, use 2 decimals.

Reply:
you mean keep the code that prints, but instead of showing the console output display everything in a table?